### ISSUE TITLE
GradientBoosting with SKORCH

### ISSUE THREAD

**Issue by francescamanni1989** [skorch-dev/skorch#787]
I would like to use skorch to implement Sklearn's GradientBoosting methods on torch based models. 
Here is my first implementation on MNIST dataset, based on a previous solved issue on ADABoost for Skorch.
<img width="308" alt="Capture" src="https://user-images.githubusercontent.com/83947886/124245087-ebb66380-db1f-11eb-8f39-77af44b6edeb.PNG">

However I keep getting this error:
<img width="446" alt="Capture" src="https://user-images.githubusercontent.com/83947886/124245221-130d3080-db20-11eb-8d53-653286f27abc.PNG">

I tried to solve it, but I don't understand exactly why there is the incompatibility.

**Comment by thomasjpfan** [comment]
scikit-learn's `GradientBoostingClassifier` does not support changing the base estimator. The implementation only works with trees.

**Comment by BenjaminBossan** [comment]
Did you mean to use [`AdaBoostingClassifier`](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.AdaBoostClassifier.html)?

**Comment by francescamanni1989** [comment]
Thanks for your prompt reply. I mean GradientBoostingClassifier. I though it could support a change on the base estimator by using Skorch since it does when using Ensemble-Pytorch.

**Comment by BenjaminBossan** [comment]
No, this does not work unfortunately. You can only change the loss, which should be a string, which is why you saw the error message above.

Given that we can't do anything about this situation on the skorch side, I'll close the issue. If you have further questions, feel free to re-open.

**Comment by francescamanni1989** [comment]
> No, this does not work unfortunately. You can only change the loss, which should be a string, which is why you saw the error message above.
> 
> Given that we can't do anything about this situation on the skorch side, I'll close the issue. If you have further questions, feel free to re-open.

Is that the same with XGB classifier? Is it possible to adapt the base estimator and interface it with Skorch?
